build.sbt

```diff
@@ -51,6 +51,19 @@ lazy val fatalWarningsSettings = Seq(
 )
 
 lazy val warningSuppression = Seq(
+  scalacOptions += "-Wconf:" + Seq(
+    "msg=APIs in chisel3.internal:s",
+    "msg=Importing from firrtl:s",
+    "msg=migration to the MLIR:s",
+    "msg=method hasDefiniteSize in trait IterableOnceOps is deprecated:s",  // replacement `knownSize` is not in 2.12
+    "msg=object JavaConverters in package collection is deprecated:s",
+    "msg=INTERNALskipFile:s",
+    "msg=undefined in comment for method cf in class PrintableHelper:s",
+    "msg=method checkLengths in class StringContext is deprecated:s"
+  ).mkString(",")
+)
+
+lazy val docWarningSuppression = Seq(
```

Comment:
`docWarningSuppression` isn't used, did you mean to just add a few things rather than fully duplicate?